tests/main/interfaces-timezone-control/task.yaml

```diff
@@ -0,0 +1,63 @@
+summary: Check that timezone interface works
+
+details: |
+    This test makes sure that a snap using the timezone-control interface
+    can access timezone information and update it.
+
+prepare: |
+    # Install a snap declaring a plug on timezone-control
+    cp $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml.orig
+    sed 's/\[\]/\[timezone-control\]/g' -i $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml
+    snap try $TESTSLIB/snaps/test-snapd-timedate-control-consumer
+
+restore: |
+    # Restore the timedate-control-consumer snap.yaml
+    cp $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml.orig $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml
+
+    # Restore the initial timezone
+    if [ -f timezone.txt ]; then
+        timedatectl set-timezone "$(cat timezone.txt)"
```

Comment:
It is possible to have a system with no timezone set, that's why you would get `n/a` in `timedatectl status`. IMO in such case it is ok to set the timezone to UTC when restoring.

Reply:
I added a workaround for that, also I filed a bug related to this. the issue is lp #1733881